### PR TITLE
PHP 8 Warnings - bunch of fixes

### PR DESCRIPTION
**Pull request by Eldenroot**
https://community.mybb.com/thread-237773.html

### CODE REVIEW

File: inc/functions_forumlist.php [hunk 1]
```diff
@@ -487,7 +487,7 @@ function get_forum_lightbulb($forum, $lastpost, $locked=0)
 	global $mybb, $lang, $db, $unread_forums;
 
 	// This forum is a redirect, so override the folder icon with the "offlink" icon.
-	if($forum['linkto'] != '')
+	if(isset($forum['linkto']) && $forum['linkto'] != '')
```
Comment (Stefan-MyBB) [inline inc/functions_forumlist.php]: You could merge both conditions with !empty.

File: printthread.php [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 // Load global language phrases
 $lang->load("printthread");
+$lang->load("archive");
```
Comment (Stefan-MyBB) [inline printthread.php]: It makes little sense to load that language file here.

**Review by Stefan-MyBB — CHANGES_REQUESTED** [review]

File: printthread.php (hunk 2 above)
Comment (Eldenroot) [reply printthread.php]: Are you sure?

**Comment by Eldenroot** [comment]
Part is fixed. Thank you!

**Comment by Eldenroot** [comment]
Is there anybody who can review this and help me a little bit with this one? Thank you!

File: printthread.php (hunk 2 above)
Comment (Stefan-MyBB) [reply printthread.php]: In any case, I would not expect this language file to be used anywhere outside the archive.

File: inc/datahandlers/post.php [hunk 3]
```diff
@@ -515,7 +515,7 @@ function verify_image_count()
 		$permissions = user_permissions($uid);
 
 		// Fetch the forum this post is being made in
-		if(!$post['fid'])
+		if(empty['fid'])
```
Comment (dvz) [inline inc/datahandlers/post.php]: A missing `(` makes the syntax checks fail

File: printthread.php (hunk 2 above)
Comment (dvz) [reply printthread.php]: Worth reviewing if both `Pages:` phrases should be included like this:
https://github.com/mybb/mybb/blob/af60e3a9e03b7e4be0d4d6a848945d026c233b0b/install/resources/mybb_theme.xml#L9894

File: inc/datahandlers/post.php (hunk 3 above)
Comment (Eldenroot) [reply inc/datahandlers/post.php]: Hmm, checked both and still invalid... reverted back.

File: printthread.php (hunk 2 above)
Comment (Eldenroot) [reply printthread.php]: I will revert this part for this PR.

File: inc/datahandlers/post.php (hunk 3 above)
Comment (dvz) [reply inc/datahandlers/post.php]: The variable name was also lost, so in the end we'll probably want: `if(empty($post['fid']))`

Comment (Eldenroot) [reply inc/datahandlers/post.php]: Thank you, this did the trick!

File: printthread.php (hunk 2 above)
Comment (Eldenroot) [reply printthread.php]: Reverted, a new ticket  should be opened.

**Pull request merged**